### PR TITLE
fixed inconsistency of IP on VR when VR is destroyed and recrea…

### PR DESCRIPTION
**Pull request by Pearl1594**
## Description
This address the issue mentioned in #3787  - Guest IP address of Shared Network VR is not persistent when VR is destroyed/recreated

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?
- Created a shared network
- Created a VM in the shared network
- destroyed the Virtual router 
- stopped and restarted the instance 
- the VR created will have the same Guest IP as it had previously

### CODE REVIEW

**Comment by rhtyd** [comment]
Regression testing and manual QA against VPC, shared and isolated network with/without redundant router. Also we need to check if L2 is affected (likely not, as there are no routers).

File: server/src/main/java/com/cloud/network/NetworkModelImpl.java [hunk 1]
```diff
@@ -2299,6 +2299,9 @@ public NicVO getPlaceholderNicForRouter(Network network, Long podId) {
                         ipv6 = _ipv6Dao.findByNetworkIdAndIp(network.getId(), nic.getIPv6Address());
                     }
                     //return nic only when its ip address belong to the pod range (for the Basic zone case)
+                    if (vlans.isEmpty()) {
+                        return nic;
+                    }
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/network/NetworkModelImpl.java]: I don't think this comment still describves the line below. Can you clarify or remove it?

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: server/src/main/java/com/cloud/network/NetworkModelImpl.java (hunk 1 above)
Comment (Pearl1594) [reply server/src/main/java/com/cloud/network/NetworkModelImpl.java]: @DaanHoogland I've moved the comment

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@Pearl1594 there is a checkstyle error. Can you check?

**Comment by Pearl1594** [comment]
@DaanHoogland Handled it

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
I will test it

**Pull request merged**